#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_password.dpatch by Niko Tyni <ntyni@iki.fi>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't try to read the password file when running as a CGI.

@DPATCH@
diff -urNad smokeping-2.0.9~/lib/Smokeping/probes/passwordchecker.pm smokeping-2.0.9/lib/Smokeping/probes/passwordchecker.pm
--- smokeping-2.0.9~/lib/Smokeping/probes/passwordchecker.pm	2005-02-27 14:43:30.000000000 +0200
+++ smokeping-2.0.9/lib/Smokeping/probes/passwordchecker.pm	2006-10-26 20:23:32.000000000 +0300
@@ -97,7 +97,7 @@
 			_example => '/some/place/secret',
 			_sub => sub {
 				my $val = shift;
-				-r $val or return "ERROR: password file $val is not readable.";
+				-r $val or $ENV{SERVER_SOFTWARE} or return "ERROR: password file $val is not readable.";
 				return undef;
 			},
 		},
